apps/prairielearn/src/components/ServerJobProgress/ServerJobProgressBars.tsx

```diff
@@ -0,0 +1,204 @@
+import { useMemo } from 'preact/compat';
+import { Alert, ProgressBar } from 'react-bootstrap';
+
+import type { JobProgress } from '../../lib/serverJobProgressSocket.shared.js';
+
+/**
+ * Displays progress information for multiple server jobs.
+ *
+ * This is used alongside the `useServerJobProgress` hook to display live progress information
+ * for server jobs, retrieved via WebSocket connections.
+ *
+ * @param params
+ *
+ * @param params.itemNames What the name of the job items are (e.g. "submissions graded", "students invited").
+ * @param params.jobsProgress Progress information for each server job.
+ *
+ * @param params.statusIcons Icons for indicating the server job status.
+ * @param params.statusIcons.inProgress Icon for in-progress jobs.
+ * @param params.statusIcons.complete Icon for completed jobs.
+ * @param params.statusIcons.failed Icon for failed jobs.
+ *
+ * @param params.statusText Text describing the server job status.
+ * @param params.statusText.inProgress Text for in-progress jobs.
+ * @param params.statusText.complete Text for completed jobs.
+ * @param params.statusText.failed Text for failed jobs.
+ *
+ * @param params.onDismissCompleteJobSequence Callback when the user dismisses a completed job progress alert. Used to remove the job from state.
+ */
+export function ServerJobsProgressInfo({
+  itemNames,
+  jobsProgress,
+  statusIcons = {},
+  statusText = {},
+  onDismissCompleteJobSequence,
+}: {
+  itemNames: string;
+  jobsProgress: JobProgress[];
+  statusIcons?: {
+    inProgress?: string;
+    complete?: string;
+    failed?: string;
+  };
+  statusText?: {
+    inProgress?: string;
+    complete?: string;
+    failed?: string;
+  };
+  onDismissCompleteJobSequence: (jobSequenceId: string) => void;
+}) {
+  const statusIconsSafe = {
+    inProgress: statusIcons.inProgress || 'bi-hourglass-split',
```

Comment:
```suggestion
    inProgress: statusIcons.inProgress ?? 'bi-hourglass-split',
```

nit (also elsewhere)